### PR TITLE
feat: switch template UI

### PR DESCRIPTION
**Pull request by nickoferrall**
Fix https://github.com/ParabolInc/parabol/issues/9087

Demo: https://www.loom.com/share/758f0aef94a943dc9ec28009125b325c

This PR won't be merged until the functionality is added in a subsequent PR.

I'm using the "Options" button from Standups. One small UI thing is the avatar list gets larger and smaller depending on the screen size whereas "options" remains the same height. As that affects Standups, I'll tackle it in a separate PR. 

I created a reusable menu & menu item component in this PR:
Screenshot 2023-11-01 at 18 51 25

### To test

- [ ] Create a retro, go to the "Reflect" phase and see "options"
- [ ] Clicking on the "Change template" menu item opens the right drawer with all the templates
- [ ] This doesn't show up in the group phase

### CODE REVIEW

File: packages/server/graphql/public/types/User.ts [hunk 1]
```diff
@@ -174,6 +174,7 @@ const User: UserResolvers = {
         ...activity,
         sortOrder: getScore(activity, teamIds)
       }))
+      .filter((activity) => !type || activity.type === type)
```
Comment (nickoferrall) [inline packages/server/graphql/public/types/User.ts:177]: I could add `type` to the dataloader (or create a second dataloader) so that we only get retrospective templates. But as this query is used in the activity library, which will probably be executed for the org before the user gets to the retro drawer, it should be more efficient to keep it as it is and get the templates from the cache.

**Review by igorlesnenko — CHANGES_REQUESTED** [review]
If we are introducing new radix components here, I would ideally make them similar to our previous radix components:

**Review by acressall — APPROVED** [review]

File: packages/client/components/MeetingOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        <MenuItem
+          value='template'
+          label='Change template'
+          onClick={handleClick}
+          isDisabled={hasReflections}
+          icon={<SwapHorizIcon />}
+        />
+      </div>
+      {tooltipPortal('You can only change the template before reflections have been added.')}
+    </Menu>
```
[the remark this answers is not in the document]
Comment (nickoferrall) [reply packages/client/components/MeetingOptions.tsx]: I prefer including `trigger` as a prop because it's always required, and it means the dev just needs to import Menu & MenuItem. I've removed the icon as a prop though

File: packages/client/ui/Menu/MenuItem.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+export const MenuItem: React.FC<MenuItemProps> = (props: MenuItemProps) => {
+  const {value, label, onClick, isDisabled, icon} = props
+
+  return (
+    <DropdownMenu.Item
+      className={`flex w-full items-center rounded-md px-4 py-3 text-sm text-slate-700 outline-none hover:bg-slate-100 focus:bg-slate-100 ${
+        isDisabled ? 'cursor-not-allowed ' : 'cursor-pointer'
+      }`}
```
[the remark this answers is not in the document]
Comment (nickoferrall) [reply packages/client/ui/Menu/MenuItem.tsx]: I've updated it to include `twMerge` and forward ref. I spent ~1 hour trying to get `forwardRadix` to work, but had problems passing the prop types down and getting the right ref type

File: packages/client/ui/Menu/Menu.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import React from 'react'
+import {twMerge} from 'tailwind-merge'
+
+interface MenuProps {
+  trigger: React.ReactNode
+  children: React.ReactNode
+}
+
+export const Menu = React.forwardRef<HTMLDivElement, MenuProps>(({trigger, children}, ref) => {
+  const contentClass = twMerge(
```
Comment (igorlesnenko) [inline packages/client/ui/Menu/Menu.tsx:11]: +1 Current usage of `twMerge` is redundant, we are using it to be able to pass our own `className`. 
Same for `MenuItem`.

https://github.com/ParabolInc/parabol/blob/27a9efbbfd680febbe8f06203dab8e53e5e1fb56/packages/client/ui/Dialog/DialogClose.tsx#L13-L15

We also passing down props to support all radix props

https://github.com/ParabolInc/parabol/blob/27a9efbbfd680febbe8f06203dab8e53e5e1fb56/packages/client/ui/Dialog/DialogClose.tsx#L18

We can leave it as is and tweak later if the need arises to not spend more time on this now

**Review by igorlesnenko — APPROVED** [review]
A note for the future: I noticed that I can add reflections while drawer is opened, so theoretically I can click on new template with already existing reflections

**Comment by igorlesnenko** [comment]
We don't need a feature flag here?

File: packages/client/ui/Menu/Menu.tsx (hunk 4 above)
Comment (nickoferrall) [reply packages/client/ui/Menu/Menu.tsx:11]: True - I've added the `className` now so twMerge is used 👍

**Comment by nickoferrall** [comment]
> We don't need a feature flag here?

Nope, I suggested that we don't use a feature flag here and just roll it out to all users. I won't merge this PR until the logic is in place though.

File: packages/client/components/MeetingOptions.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        <div onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+          <TooltipTrigger asChild>
+            <MenuItem onClick={handleClick} isDisabled={hasReflections}>
+              <div className='mr-3 flex text-slate-700'>{<SwapHorizIcon />}</div>
+              Change template
+            </MenuItem>
+          </TooltipTrigger>
+        </div>
+        <TooltipContent>
+          {'You can only change the template before reflections have been added.'}
```
Comment (Dschoordsch) [inline packages/client/components/MeetingOptions.tsx:55]: +1 I would rephrase this slightly to "You can only change the template if no reflections have been added." because it also works if you remove them again.

**Review by Dschoordsch — APPROVED** [review]
